### DISCUSSION TITLE
Question about version of Yosys used by OpenLane

### DISCUSSION THREAD

**Discussion by titan73** [The-OpenROAD-Project/OpenLane#1223] [Q&A]
Hi,

The version used is the cfe940a98b08f1a5d08fb44427db155ba1f18b62 from YosysHQ which dates to 2021 december the 26th, a little before Yosys 0.13 was released.
Why is the version used so old? And why it is not using an official release?
If Yosys is from YosysHQ, what is the purpose of the Yosys repo in OpenROAD project?

Thanks in advance
Chris

**Comment by maliberty** [comment]
I can't speak to the OL version but we tend to fork our upstream repos for stability.  In the past we have had upstream repos disappear or rewrite their history causing us trouble.  Sometime we might fix a bug locally without having to wait for the upstream to accept the PR as well.

**Reply by titan73** [reply]
Yeah makes sense. This answer the second question. Thanks!
If anyone knows about why Yosys version is old, I'll be glad to hear.

**Comment by mithro** [comment]
@mkkassem - This really should be fixed.

**Comment by donn** [comment]
@mithro There's a reason it has *not* been "fixed", and that is because there inevitably are some regressions. We do not have the same direct access to the Yosys team that we do the OpenROAD, Magic and Netgen teams, so getting issues fixed in a timely manner is considerably more difficult.

Do you have a specific need for an up-to-date version of Yosys?

**Reply by mithro** [reply]
The longer you go between updating Yosys versions the **more** regressions there will be and they will also be **considerably** harder to fix. 

If you find a regression moments after they land into the Yosys tree then it is generally very easy to get them fixed.

With a small team, the only hope you have of not suffering massive unfixable regressions is to stay on the latest head version.
